go.mod

```diff
@@ -1,6 +1,6 @@
 module github.com/prometheus/common
 
-go 1.21
+go 1.23.0
```

Comment:
Would this force golang to set their minimum go version to 1.23 as well? I believe we're still in 1.22 over there 🤔

Reply:
Yes, unfortunately, the upstream Go maintainers (`golang.org/x/oauth2`) have set this as a minimum requirement.

There's now a upstream Go bot auto-updating this: https://github.com/golang/oauth2/commit/3f78298beea38fb76a3fbca33e3056f4b7eb5502

We'll have to reconsider our support policy for client_golang versions. :crying_cat_face: